torchgeo/datasets/flair2.py

```diff
@@ -0,0 +1,762 @@
+            # Format the glob of sentinel `SEN_{0}.npy` to match the actual file name
+            # in other cases, where it is not a format string, the glob will be the same
+            if not glob.glob(files_glob.format(''), recursive=True):
+                to_extract.append(dir_name)
+
+        if not to_extract:
+            print('Data has been downloaded and extracted already...')
+            return
+
+        # Deepcopy files to be extracted and check wether the zip is downloaded
+        to_download = list(map(lambda x: x, to_extract))
+        for candidate in to_extract:
+            zipfile = os.path.join(self.root, f'{candidate}.zip')
+            if glob.glob(zipfile):
+                print(f'Extracting: {candidate}')
+                self._extract(candidate)
+                to_download.remove(candidate)
+
+        # Check if there are still files to download
+        if not to_download:
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise DatasetNotFoundError(self)
+
+        print('Downloading: ', to_download)
+        for candidate in to_download:
+            self._download(candidate)
+            self._extract(candidate)
+
+    def _download(self, url: str, suffix: str = '.zip') -> None:
+        """Download the dataset."""
+        download_url(
+            os.path.join(self.url_prefix, f'{url}{suffix}'),
+            self.root,
+            md5=self.md5s.get(url, None) if self.checksum else None,
+        )
+        # FIXME: Why is download_url not checking integrity (tests run through)?
+        assert check_integrity(
+            os.path.join(self.root, f'{url}{suffix}'),
+            self.md5s.get(url, None) if self.checksum else None,
+        )
+
+    def _extract(self, file_path: str) -> None:
+        """Extract the dataset."""
+        assert isinstance(self.root, str | os.PathLike)
+        zipfile = os.path.join(self.root, f'{file_path}.zip')
+        extract_archive(zipfile)
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: str | None = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample return by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional suptitle to use for figure
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+
+        def normalize_plot(tensor: Tensor) -> Tensor:
+            """Normalize the plot."""
+            return (tensor - tensor.min()) / (tensor.max() - tensor.min())
+
+        rgb_indices = [self.all_bands.index(band) for band in self.rgb_bands]
+        # Check if RGB bands are present in self.bands
+        if not all([band in self.bands for band in self.rgb_bands]):
```

Comment:
The Codecoverage is indicating that the RGB Band Missing is not being hit, so I think you just need to add a separate plot test similar to https://github.com/microsoft/torchgeo/blob/2f3e8fda8f1d0d585ac480696434e5c69427cf8a/tests/datasets/test_eurosat.py#L110 for example.